src/ragapp/backend/middleware.py

```diff
@@ -0,0 +1,72 @@
+import os
+import time
+
+import jwt
+from fastapi import HTTPException, Request, status
+from fastapi.responses import Response
+from jwt import InvalidTokenError
+from sqlalchemy.orm import Session
+
+from backend.controllers.chat_request import (
+    get_user_chat_request_count,
+    update_user_chat_request_count,
+)
+from backend.database import DB
+
+JWT_COOKIE_NAME = "Authorization"  # The name of the cookie that stores the JWT token
+JWT_USER_ID_CLAIM = "preferred_username"  # The claim in the JWT token that stores the user ID or user name
+CHAT_REQUEST_LIMIT_THRESHOLD = int(os.environ.get("CHAT_REQUEST_LIMIT_THRESHOLD", 5))
+
+
+async def request_limit_middleware(request: Request) -> Response:
+    time_frame = _get_time_frame()
+    user_id = _extract_user_id_from_request(request)
+    db: Session = next(DB.get_session())
+    request_count = get_user_chat_request_count(db, user_id, time_frame)
```

Comment:
how about using OO?
```suggestion
    service = UserChatService()
    request_count = service.get_request_count(user_id, time_frame)
```
`UserChatService` can then look up the db session

Reply:
we might then also move this service to a `services` folder

Reply:
I moved to `UserChatService` but haven't removed the db params yet. The reason is i think we will soon need to add management api to clear/refresh the rate limit - and the db session is a dependency for the api request which is injected automatically.

Reply:
how about adding the session to the constructor of the `UserChatService`?
if it's not passed `DB.get_session()` is used per default - then you can still pass the session if needed and we would have here:
```
service = UserChatService()
request_count = service.get_request_count(user_id, time_frame)
```

Reply:
Ok, so i just moved the db session to the UserChatService. I'll add the session parameter once we implement new feature that need to reuse the available db session :D